lib/matplotlib/_mathtext.py

```diff
@@ -2627,6 +2628,9 @@ def subsuper(self, s, loc, toks):
             state.font, state.fontsize, state.dpi)
         xHeight = state.font_output.get_xheight(
             state.font, state.fontsize, state.dpi)
+        capHeight = state.font_output.get_metrics(
+            state.font, mpl.rcParams['mathtext.default'],
+            'H', state.fontsize, state.dpi).height
```

Comment:
I assume you need this because you don't really have the nucleus size? I wonder if there is a way to make that available, but I don't really know `pyparsing` and whether that is possible.

Reply:
The `nucleus` variable is available right there, but perhaps I misunderstood your question?

Reply:
The nucleus is accessible with the change mentioned in the original post. I used `capHeight` as a yardstick to determine the shifts.

To be honest, I had completely forgotten about this PR. Having taken another look, I feel it can be improved further (without using hacky things like `capHeight`) now that I have learnt a few more things about Mathtext. Just that I'm really busy with college and haven't found time to work on this.